addons/account_peppol/models/res_company.py

```diff
@@ -86,20 +82,16 @@ def _sanitize_peppol_phone_number(self):
             "For example: +32123456789, where +32 is the country code.\n"
             "Currently, only European countries are supported.")
 
-        if not phonenumbers:
-            raise ValidationError(error_message)
-
-        phone_number = self.account_peppol_phone_number
-        if not phone_number.startswith('+'):
-            phone_number = f'+{phone_number}'
-
         try:
-            phone_nbr = phonenumbers.parse(phone_number)
-        except phonenumbers.phonenumberutil.NumberParseException:
-            raise ValidationError(error_message)
-
-        country_code = phonenumbers.phonenumberutil.region_code_for_number(phone_nbr)
-        if country_code not in PHONE_ALLOWED_COUNTRIES or not phonenumbers.is_valid_number(phone_nbr):
+            phone_nbr = self._phone_format_number(
+                self.account_peppol_phone_number,
+                raise_exception=True
+            )
+        except UserError as err:
+            raise ValidationError(error_message) from err
+
+        country_code = phone_validation.phone_get_region_code(phone_nbr)
```

Comment:
Note that the "phone_parse" version raises more exceptions than "phonenumbers.parse(phone_number)".

Maybe this should be wrapped into a try/catch to return the pre-formatted "error_message" like above?